entviz.go

```diff
@@ -3,23 +3,83 @@ package entviz
 import (
 	"bytes"
 	_ "embed"
+	"encoding/json"
 	"html/template"
+
 	"io/ioutil"
-	"strings"
 
 	"entgo.io/ent/entc/gen"
 )
 
-//go:embed template.html
+type (
+	jsGraph struct {
+		Nodes []jsNode `json:"nodes"`
+		Edges []jsEdge `json:"edges"`
+	}
+
+	jsNode struct {
+		ID     string    `json:"id"`
+		Fields []jsField `json:"fields"`
+	}
+
+	jsEdge struct {
+		From  string `json:"from"`
+		To    string `json:"to"`
+		Label string `json:"label"`
+	}
+
+	jsField struct {
+		Name string `json:"name"`
+		Type string `json:"type"`
+	}
+)
+
+// toJsGraph converts ent's graph into json serializable struct
+func toJsGraph(g *gen.Graph) *jsGraph {
+	var jsGraph = &jsGraph{}
+	for _, n := range g.Nodes {
+		node := jsNode{ID: n.Name}
+
+		for _, f := range n.Fields {
+			node.Fields = append(node.Fields, jsField{
+				Name: f.Name,
+				Type: f.Type.String(),
+			})
+		}
+		jsGraph.Nodes = append(jsGraph.Nodes, node)
+
+		for _, e := range n.Edges {
+			if e.IsInverse() {
+				continue
+			}
+			jsGraph.Edges = append(jsGraph.Edges, jsEdge{
+				From:  n.Name,
+				To:    e.Type.Name,
+				Label: e.Name,
+			})
+		}
+
+	}
+	return jsGraph
+}
+
+//go:embed viz.html
 var tmplhtml string
 
+// VisualizeSchema is an ent's hook that generates static html that visualizes the schema graph.
```

Comment:
```suggestion
// VisualizeSchema is an ent hook that generates a static html page that visualizes the schema graph.
```

I'm not sure the apostrophe is needed here.